packages/backend-app-api/src/wiring/BackendInitializer.ts

```diff
@@ -336,6 +336,21 @@ export class BackendInitializer {
       await this.#serviceRegistry.get(coreServices.rootLogger, 'root'),
     );
 
+    const rootConfig = await this.#serviceRegistry.get(
+      coreServices.rootConfig,
+      'root',
+    );
+
+    const startupOptions =
+      rootConfig
+        ?.getOptionalConfig('backend.startup')
+        ?.get<BackendStartupOptions>() ?? {};
+
+    // Gather backend pluginIds marked as optional, since these should not cause a startup failure
+    const optionalBackends = Object.entries(startupOptions)
+      .filter(([_, value]) => value?.optional)
+      .map(([key]) => key);
```

Comment:
I think it would be nice to encapsulate this a little bit, just something simple like `const isPluginOptional = this.#getPluginOptionalityPredicate()`?